service/worker/workerdeployment/workflow.go

```diff
@@ -127,12 +127,33 @@ func (d *WorkflowRunner) updateVersionSummary(summary *deploymentspb.WorkerDeplo
 }
 
 func (d *WorkflowRunner) run(ctx workflow.Context) error {
-	if d.GetState().GetCreateTime() == nil {
-		if d.State == nil {
-			d.State = &deploymentspb.WorkerDeploymentLocalState{}
+	// TODO(carlydf): remove verbose logging
+	d.logger.Info("Raw workflow state at start",
+		"state_nil", d.State == nil,
+		"create_time_nil", d.GetState().GetCreateTime() == nil,
+		"routing_config_nil", d.GetState().GetRoutingConfig() == nil,
+		"raw_state", d.State,
+		"workflow_id", workflow.GetInfo(ctx).WorkflowExecution.ID,
+		"run_id", workflow.GetInfo(ctx).WorkflowExecution.RunID)
+
+	if d.State == nil {
+		d.State = &deploymentspb.WorkerDeploymentLocalState{}
+	}
+
+	// Defensive check: ensure RoutingConfig is never nil
+	if d.State.RoutingConfig == nil {
+		d.logger.Warn("RoutingConfig was nil, initializing with default values",
+			"workflow_id", workflow.GetInfo(ctx).WorkflowExecution.ID,
+			"run_id", workflow.GetInfo(ctx).WorkflowExecution.RunID)
+		d.State.RoutingConfig = &deploymentpb.RoutingConfig{CurrentVersion: worker_versioning.UnversionedVersionId}
+		// Update memo to persist the fixed state
+		if err := d.updateMemo(ctx); err != nil {
```

Comment:
omg maybe the replay test is catching the moment when RoutingConfig is nil, because thats where we're generating a new update memo command